src/App.jsx

```diff
@@ -1,13 +1,214 @@
-
 import './App.css'
+import {useState} from "react";
+import FruitCounter from "./assets/components/FruitCounter/FruitCounter.jsx";
+import Button from "./assets/components/Button/Button.jsx";
+import FormInputField from "./assets/components/FormInputField/FormInputField.jsx";
 
 function App() {
+    const [counterState, setCounterState] = useState({
+        strawberryCounter: 0,
+        bananaCounter: 0,
+        appleCounter: 0,
+        kiwiCounter: 0
+    });
+    const [formState, setFormState] = useState({
+        firstName: '',
+        lastName: '',
+        age: 0,
+        postCode: '',
+        deliveryFrequency: 'weekly',
+        timeSlot: 'day',
+        comments: '',
+        terms: false
+
+    });
+
+    // handleClick zorgt voor het op- en aftellen van de counters
+    // waarde -1 of +1 wordt meegegeven afhankelijk van de button
+    function handleClick(counterName, newValue) {
+        setCounterState({
+            ...counterState,
+            [counterName]: (counterState[counterName] + newValue)
+        })
+    }
+
+    function resetCounters() {
+        // Dynamisch resetten van de counters - ongeacht het aantal in het state-object,
+        // Object.keys leest het object uit in een array om overheen te mappen
+        // Object.fromEntries maakt van de array weer een object zoals de state terugverwacht
+        setCounterState((prevState) =>
+            Object.fromEntries(Object.keys(prevState).map((key) => [key, 0]))
```

Comment:
Interessante oplossing! 

Wat je ook zo kunnen doen is het object variabel maken, zo kan je die ook nog eens vaker gebruiken.

vb:

`const intialFruitCounter = {  strawberryCounter: 0,
        bananaCounter: 0,
        appleCounter: 0,
        kiwiCounter: 0
    }`

Vervolgens kan je dit object als initialState plaatsen:

`const [counterState, setCounterState] = useState(initialFruitCounter)`

En nu kan je de reset heel simpel uitvoeren:

`function resetCounters(){
setCounterState(initialFruitCounter)
}`